rtil/src/native/uworld.rs

```diff
@@ -158,9 +164,229 @@ impl UWorld {
+        time_of_day.get_field("CurrentMinute").unwrap::<&Cell<f32>>().set(time);
+    }
+
+    pub fn set_cloud_speed(speed: f32) {
+        UeScope::with(|scope| {
+            let object = scope.get(CLOUDS_INDEX.get().unwrap());
+            let fun: extern_fn!(fn( this: *const c_void, name: FName, value: f32 ))
+                = unsafe { mem::transmute(UMATERIALINSTANCEDYNAMIC_SETSCALARPARAMETERVALUE.load(Ordering::SeqCst)) };
+            fun(object.as_ptr() as *const c_void, FName::from("Cloud speed"), speed);
+        })
+    }
+
+    pub fn set_outro_time_dilation(dilation: f32) {
+        UeScope::with(|scope| {
+            let object = scope.get(JUMP6_INDEX.get().unwrap());
+            object.get_field("OutroDilation").unwrap::<&Cell<f32>>().set(dilation);
+        })
+    }
+
+    pub fn set_outro_dilated_duration(duration: f32) {
+        UeScope::with(|scope| {
+            let object = scope.get(JUMP6_INDEX.get().unwrap());
+            object.get_field("OutroDilatedDuration").unwrap::<&Cell<f32>>().set(duration);
+        })
+    }
+
+    pub fn set_kill_z(kill_z: f32) {
+        let obj = unsafe { ObjectWrapper::new(UWorld::get_global() as *mut UObject) };
+        let persistent_level = obj.get_field("PersistentLevel").unwrap::<ObjectWrapper>();
+        let world_settings = persistent_level.get_field("WorldSettings").unwrap::<ObjectWrapper>();
+        world_settings.get_field("KillZ").unwrap::<&Cell<f32>>().set(kill_z);
+    }
+
+    pub fn set_reflection_render_scale(render_scale: i32) {
+        let obj = unsafe { ObjectWrapper::new(UMyGameInstance::get_umygameinstance() as *mut UObject) };
+        let world_references = obj.get_field("WorldReferences").unwrap::<ObjectWrapper>();
+        let water = world_references.get_field("Water").unwrap::<ObjectWrapper>();
+        let component = water.get_field("PlanarReflectionComponent").unwrap::<ObjectWrapper>();
+        component.get_field("ScreenPercentage").unwrap::<&Cell<i32>>().set(render_scale);
+    }
+
+    pub fn enable_fog(value: bool) {
+        let obj = unsafe { ObjectWrapper::new(UMyGameInstance::get_umygameinstance() as *mut UObject) };
+        let world_references = obj.get_field("WorldReferences").unwrap::<ObjectWrapper>();
+        let time_of_day = world_references.get_field("TimeOfDay").unwrap::<ObjectWrapper>();
+        let fog = time_of_day.get_field("Fog").unwrap::<ObjectWrapper>();
+        let component = fog.get_field("Component").unwrap::<ObjectWrapper>();
+        if value {
+            component.get_field("FogDensity").unwrap::<&Cell<f32>>().set(0.05);
+            component.get_field("DirectionalInscatteringExponent").unwrap::<&Cell<f32>>().set(0.);
+        } else {
+            component.get_field("FogDensity").unwrap::<&Cell<f32>>().set(0.);
+            component.get_field("DirectionalInscatteringExponent").unwrap::<&Cell<f32>>().set(0.);
+        }
```

Comment:
Is it a bug that `DirectionalInscatteringExponent` is always set to 0?

Reply:
Nope, it only needs to be updated to any value for the changes to take effect, but I guess I can change it to `1` just in case it happens to be `0` when I'm changing it.

Reply:
If it needs to be updated to the same value in both cases, you can pull it out of the if-else and only have the statement once after the if-else.